packages/data-service/src/connection-storage.ts

```diff
@@ -0,0 +1,95 @@
+import { ConnectionOptions } from './connection-options';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { ConnectionCollection } = require('mongodb-connection-model');
+
+import {
+  convertConnectionModelToOptions,
+  convertConnectionOptionsToModel,
+  AmpersandMethodOptions,
+} from './legacy-connection-model';
+
+type ConnectionOptionsWithRequiredId = ConnectionOptions &
+  Required<Pick<ConnectionOptions, 'id'>>;
+
+import { validate as uuidValidate } from 'uuid';
+
+export class ConnectionStorage {
+  /**
+   * Loads all the ConnectionOptions currently stored.
+   *
+   * @returns Promise<ConnectionOptions[]>
+   */
+  async loadAll(): Promise<ConnectionOptions[]> {
+    const connectionCollection = new ConnectionCollection();
+    const fetchConnectionModels = promisifyAmpersandMethod(
+      connectionCollection.fetch.bind(connectionCollection)
+    );
+
+    await fetchConnectionModels();
+    return connectionCollection.map(convertConnectionModelToOptions);
+  }
+
+  /**
+   * Inserts or replaces a ConnectionOptions object in the storage.
+   *
+   * The ConnectionOptions object must have an id set to a string
+   * matching the uuid format.
+   *
+   * @param connectionOptions - The ConnectionOptions object to be saved.
+   */
+  async save(
+    connectionOptions: ConnectionOptionsWithRequiredId
+  ): Promise<void> {
+    if (!connectionOptions.id) {
+      throw new Error('id is required');
+    }
+
+    if (!uuidValidate(connectionOptions.id)) {
+      throw new Error('id must be a uuid');
+    }
+
+    const model = await convertConnectionOptionsToModel(connectionOptions);
+    model.save();
```

Comment:
There is no way to make sure the write actually succeeds, nor to catch a failure.